keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,147 @@
+
+
+def compute_word_piece_vocabulary(
+    data,
+    vocabulary_size,
+    vocabulary_output_file=None,
+    lowercase=True,
+    strip_accents=True,
+    split=True,
+    suffix_indicator="##",
+    reserved_tokens=["[PAD]", "[CLS]", "[SEP]", "[UNK]", "[MASK]"],
+):
+    r"""A utility to train a Word Piece vocabulary.
+
+    Trains a Word Piece vocabulary from an input dataset or a list of filenames.
+
+    For custom data loading and pretokenization (`split=False`), the input
+    `data` should be a `tf.data.Dataset`. If `data` is a list of filenames,
+    the file format is required to be plain text files, and the text would be
+    read in line by line during training.
+
+    Args:
+        data: A tf.data.Dataset, or a list of filenames.
+        vocabulary_size: The maximum size of a vocabulary to be trained.
+        vocabulary_output_file: The location to write a vocabulary file.
+        lowercase: If true, the input text will be first lowered before
+            tokenization.
+        strip_accents: If true, all accent marks will be removed from text
+            before tokenization.
+        split: If true, input will be split on whitespace and punctuation
+            marks, and all punctuation marks will be kept as tokens. If false,
+            input should be split ("pre-tokenized") before calling the
+            tokenizer, and passed as a dense or ragged tensor of whole words.
+            `split` is required to be`True` when `data` is a list of filenames.
+        suffix_indicator: The characters prepended to a wordpiece to indicate
+            that it is a suffix to another subword.
+        reserved_tokens: A list of tokens that must be included in the vocabulary.
+
+    Returns:
+        Returns a list of vocabulary terms.
+
+    Examples:
+
+    Basic Usage (from Dataset).
+    >>> inputs = tf.data.Dataset.from_tensor_slices(["bat sat pat mat rat"])
+    >>> vocab = compute_word_piece_vocabulary(inputs, 9, reserved_tokens=["[UNK]"])
+    >>> vocab
+    ['[UNK]', 'a', 'b', 'm', 'p', 'r', 's', 't', '##at']
+    >>> tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(vocabulary=vocab, oov_token="[UNK]")
+    >>> outputs = inputs.map(tokenizer.tokenize)
+    >>> for x in outputs: print(x)
+    tf.Tensor([2 8 6 8 4 8 3 8 5 8], shape=(10,), dtype=int32)
+
+    Basic Usage (from filenames).
+    >>> with open("test.txt", "w+") as f: f.write("bat sat pat mat rat\n")
+    >>> inputs = ["test.txt"]
+    >>> vocab = compute_word_piece_vocabulary(inputs, 9, reserved_tokens=["[UNK]"])
+
+    Custom Split Usage (from Dataset).
+    >>> def normalize_and_split(x):
+    ...     "Strip punctuation and split on whitespace."
+    ...     x = tf.strings.regex_replace(x, r"\p{P}", "")
+    ...     return tf.strings.split(x)
+    >>> inputs = tf.data.Dataset.from_tensor_slices(["bat sat: pat mat rat.\n"])
+    >>> split_inputs = inputs.map(normalize_and_split)
+    >>> vocab = compute_word_piece_vocabulary(
```

Comment:
same here and elsewhere, don't set reserved tokens

Reply:
I mainly set the reserved tokens to shorten the output (so it doesn't display too many things), but I can remove it and still fit it in a line.